ansible/roles/ocp4-workload-rhtr-iot-managed/tasks/pre_workload.yml

```diff
@@ -0,0 +1,67 @@
+---
+# Implement your Pre Workload deployment tasks here
+
+- name: Ensuring AWS region is defined
+  fail:
+    msg: "This workload requires aws_region to be defined. Exiting..."
+  when: (aws_region is not defined) and (ocp_workload_test is undefined)
+
+- name: usercount debug
+  debug:
+    msg: "Debugging num_users {{ num_users }}"
+
+- name: create usernames
+  set_fact:
+    users: "{{ users | default([]) + ['user'+item | string] }}"
+  loop: "{{ range(1,((num_users | int) + 1)) | list }}"
+
+# Figure out paths
+- name: extract api_url
+  shell: oc whoami --show-server
+  register: api_url_r
+
+- name: set the master
+  set_fact:
+    master_url: "{{ api_url_r.stdout | trim }}"
+
+- name: extract console_url
+  command: oc whoami --show-console
+  register: console_url_r
+
+- name: set the console
+  set_fact:
+    console_url: "{{ console_url_r.stdout | trim }}"
+
+- name: extract route_subdomain
+  k8s_facts:
```

Comment:
Please move this to `k8s_info`. Facts is deprecated and info is available in 2.9+